Sources/TLS/Socket.swift

```diff
@@ -17,20 +18,31 @@ public final class Stream {
          - parameter context: Re-usable SSL.Context in either Client or Server mode
          - parameter descriptor: The file descriptor from an unsecure socket already created.
     */
-    public init(context: Context, certificates: Certificates, socket: Int32) throws {
+    public init(context: Context, certificates: Certificates, config: TLSConfig, socket: Int32) throws {
         self.context = context
+        self.config = config
         cConfig = tls_config_new()
         self.socket = socket
 
         self.certificates = certificates
```

Comment:
I'm not sure. `Config` is something coming from the end user, whereas `Context` is created to keep state for the algorithm, it's basically just an implementation detail. I don't think we should create `Context` until this initializer, but `Config` is something we'll want created during the server/client setup, to store our preferences about the TLS connection.